pkg/sloop/ingress/kubewatcher.go

```diff
@@ -81,77 +100,149 @@ func NewKubeWatcherSource(kubeClient kubernetes.Interface, outChan chan typed.Ku
 	i.informerFactory.Apps().V1().Deployments().Informer().AddEventHandler(i.getEventHandlerForResource("Deployment"))
+	i.informerFactory.Apps().V1().ReplicaSets().Informer().AddEventHandler(i.getEventHandlerForResource("ReplicaSet"))
 	i.informerFactory.Apps().V1().StatefulSets().Informer().AddEventHandler(i.getEventHandlerForResource("StatefulSet"))
+
 	i.informerFactory.Core().V1().ConfigMaps().Informer().AddEventHandler(i.getEventHandlerForResource("ConfigMap"))
 	i.informerFactory.Core().V1().Endpoints().Informer().AddEventHandler(i.getEventHandlerForResource("Endpoint"))
+	i.informerFactory.Core().V1().Events().Informer().AddEventHandler(i.getEventHandlerForResource("Event"))
 	i.informerFactory.Core().V1().Namespaces().Informer().AddEventHandler(i.getEventHandlerForResource("Namespace"))
 	i.informerFactory.Core().V1().Nodes().Informer().AddEventHandler(i.getEventHandlerForResource("Node"))
 	i.informerFactory.Core().V1().PersistentVolumeClaims().Informer().AddEventHandler(i.getEventHandlerForResource("PersistentVolumeClaim"))
 	i.informerFactory.Core().V1().PersistentVolumes().Informer().AddEventHandler(i.getEventHandlerForResource("PersistentVolume"))
 	i.informerFactory.Core().V1().Pods().Informer().AddEventHandler(i.getEventHandlerForResource("Pod"))
 	i.informerFactory.Core().V1().Services().Informer().AddEventHandler(i.getEventHandlerForResource("Service"))
 	i.informerFactory.Core().V1().ReplicationControllers().Informer().AddEventHandler(i.getEventHandlerForResource("ReplicationController"))
-	i.informerFactory.Extensions().V1beta1().DaemonSets().Informer().AddEventHandler(i.getEventHandlerForResource("DaemonSet"))
-	i.informerFactory.Extensions().V1beta1().ReplicaSets().Informer().AddEventHandler(i.getEventHandlerForResource("ReplicaSet"))
+
 	i.informerFactory.Storage().V1().StorageClasses().Informer().AddEventHandler(i.getEventHandlerForResource("StorageClass"))
-	i.informerFactory.Core().V1().Events().Informer().AddEventHandler(i.getEventHandlerForResource("Event"))
+
 	i.informerFactory.Start(i.stopChan)
 }
 
 func (i *kubeWatcherImpl) startCustomInformers(masterURL string, kubeContext string) error {
+
 	clientCfg := getConfig(masterURL, kubeContext)
 	kubeCfg, err := clientCfg.ClientConfig()
 	if err != nil {
 		return errors.Wrap(err, "failed to read config while starting custom informers")
 	}
 
-	crdList, err := getCrdList(kubeCfg)
+	crdClient, err := newCrdClient(kubeCfg)
+	if err != nil {
+		return errors.Wrap(err, "failed to instantiate client for querying CRDs")
+	}
+	crdList, err := getCrdList(crdClient)
 	if err != nil {
 		return err
 	}
 
+	glog.Infof("Found %d CRD definitions", len(crdList))
 	dynamicClient, err := dynamic.NewForConfig(kubeCfg)
 	if err != nil {
 		return errors.Wrap(err, "failed to instantiate client for custom informers")
 	}
-
-	f := dynamicinformer.NewFilteredDynamicSharedInformerFactory(dynamicClient, i.resync, "", nil)
+	existing := i.pullCrdInformers()
```

Comment:
i would rename it as crdInformers

Reply:
I wanted to express that there is more happening then just accessing the collection. The implementation of `pullCrdInformers` provides the collection of CRD informers, but leaves the in-object collection empty. This preps for the next step of refilling the collection with active/new CRD informers